src/vdom/diff.js

```diff
@@ -43,15 +43,19 @@ export function diff(dom, vnode, context, mountAll, parent, componentRoot) {
 		// hydration is indicated by the existing element to be diffed not having a prop cache
 		hydrating = dom!=null && !(ATTR_KEY in dom);
 	}
-
-	let ret = idiff(dom, vnode, context, mountAll, componentRoot);
-
-	// append the element if its a new parent
-	if (parent && ret.parentNode!==parent) parent.appendChild(ret);
+	let ret;
+	try {
+		ret = idiff(dom, vnode, context, mountAll, componentRoot);
+
+		// append the element if its a new parent
+		if (parent && ret.parentNode !== parent) parent.appendChild(ret);
+	} finally {
```

Comment:
Are the additional changes here needed, or just the try/finally?

Reply:
What do you mean by additional changes? I need to adjust `diffLevel` and `hydrating` in the finally block so that it will be executed even when an error is thrown. Aside from that, there are no additional changes.

Reply:
Guess `ret` should be initialized though, to prevent other bugs

Reply:
For clarification:
For this change, there are 2 important scenarios how the code will execute:
1. The `render` works as expected and the code is executed as before. `diffLevel` and `hydrating` are calculated in a different spot but the result is the same.
2. The `render` throws an error, which leads to `idiff` throwing an error which leads to this function aborting execution and not executing anything below line 48 - except for the finally block. This is necessary to get preact into a sane state again before leaving the function.

So I don't think it is necessary to initialise `ret`. It will be initialised in the next line and if `idiff` throws an error then  `ret` won't be used anymore.

Reply:
noice